eng/requirements.txt

```diff
@@ -1,5 +0,0 @@
-# This is for CI requirements or dev only
-pyright==1.1.391
-pylint==3.2.7
-tox==4.23.2
-mypy==1.10.1
```

Comment:
Eng is only needed by pipeline for development, so there is no need to keep 2 files that requirements.txt and dev_requirements.txt, then I combined them into 1 file: `dev_requirements.txt`.